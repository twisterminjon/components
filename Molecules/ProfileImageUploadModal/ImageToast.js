import React from 'react'
import PropTypes from 'prop-types'

import ModalToast from '../../Atoms/ModalToast/ModalToast'
import Button from '../../Atoms/Button/Button'
import ButtonGhost from '../../Atoms/ButtonGhost/ButtonGhost'
import Stack from '../../Atoms/Stack/Stack'

import './ImageToast.css'

ImageToast.propTypes = {
  /** Can show a loader on the remove button */
  loadingRemove: PropTypes.bool,

  /** Can be shown or hidden */
  show: PropTypes.bool.isRequired,

  /** Called after the select image action */
  onSelectImage: PropTypes.func.isRequired,

  /** Called after ImageToast item has been selected */
  onRemoveImage: PropTypes.func.isRequired,

  /** Called after ImageToast item has been selected */
  onClose: PropTypes.func.isRequired,
}
ImageToast.defaultProps = {
  loadingRemove: false,
}

export default function ImageToast({ loadingRemove, show, onSelectImage, onRemoveImage, onClose }) {
  return (
    <ModalToast show={show}>
      <Stack className="imagetoast" gap={32}>
        <Button fluid onClick={onSelectImage} data-testid="image-toast-choose">
          Choose photo
        </Button>
        <Button
          fluid
          onClick={onRemoveImage}
          data-testid="image-toast-remove"
          loading={loadingRemove}
          disabled={loadingRemove}>
          Remove photo
        </Button>
        <ButtonGhost fluid onClick={onClose} data-testid="image-toast-close">
          Close
        </ButtonGhost>
      </Stack>
    </ModalToast>
  )
}
