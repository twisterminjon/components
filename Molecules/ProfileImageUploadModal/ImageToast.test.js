import React from 'react'
import ImageToast from './ImageToast'

describe('ImageToast', () => {
  const mockFun = jest.fn()

  it('it matches the snapshot when shown', () => {
    const wrapper = window.shallow(
      <ImageToast show={true} loadingRemove={false} onSelectImage={mockFun} onRemoveImage={mockFun} onClose={mockFun} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('it matches the snapshot when hidden', () => {
    const wrapper = window.shallow(
      <ImageToast show={false} onSelectImage={mockFun} onRemoveImage={mockFun} onClose={mockFun} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('it matches the snapshot when loading', () => {
    const wrapper = window.shallow(
      <ImageToast show={true} loadingRemove={true} onSelectImage={mockFun} onRemoveImage={mockFun} onClose={mockFun} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
