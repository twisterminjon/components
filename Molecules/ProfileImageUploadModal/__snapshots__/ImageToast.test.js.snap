// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageToast it matches the snapshot when hidden 1`] = `
<ModalToast
  show={false}
>
  <Stack
    className="imagetoast"
    gap={32}
    style={Object {}}
  >
    <Button
      data-testid="image-toast-choose"
      fluid={true}
      onClick={[MockFunction]}
      style={Object {}}
      type="submit"
    >
      Choose photo
    </Button>
    <Button
      data-testid="image-toast-remove"
      disabled={false}
      fluid={true}
      loading={false}
      onClick={[MockFunction]}
      style={Object {}}
      type="submit"
    >
      Remove photo
    </Button>
    <ButtonGhost
      className=""
      data-testid="image-toast-close"
      fluid={true}
      onClick={[MockFunction]}
      red={false}
      style={Object {}}
    >
      Close
    </ButtonGhost>
  </Stack>
</ModalToast>
`;

exports[`ImageToast it matches the snapshot when loading 1`] = `
<ModalToast
  show={true}
>
  <Stack
    className="imagetoast"
    gap={32}
    style={Object {}}
  >
    <Button
      data-testid="image-toast-choose"
      fluid={true}
      onClick={[MockFunction]}
      style={Object {}}
      type="submit"
    >
      Choose photo
    </Button>
    <Button
      data-testid="image-toast-remove"
      disabled={true}
      fluid={true}
      loading={true}
      onClick={[MockFunction]}
      style={Object {}}
      type="submit"
    >
      Remove photo
    </Button>
    <ButtonGhost
      className=""
      data-testid="image-toast-close"
      fluid={true}
      onClick={[MockFunction]}
      red={false}
      style={Object {}}
    >
      Close
    </ButtonGhost>
  </Stack>
</ModalToast>
`;

exports[`ImageToast it matches the snapshot when shown 1`] = `
<ModalToast
  show={true}
>
  <Stack
    className="imagetoast"
    gap={32}
    style={Object {}}
  >
    <Button
      data-testid="image-toast-choose"
      fluid={true}
      onClick={[MockFunction]}
      style={Object {}}
      type="submit"
    >
      Choose photo
    </Button>
    <Button
      data-testid="image-toast-remove"
      disabled={false}
      fluid={true}
      loading={false}
      onClick={[MockFunction]}
      style={Object {}}
      type="submit"
    >
      Remove photo
    </Button>
    <ButtonGhost
      className=""
      data-testid="image-toast-close"
      fluid={true}
      onClick={[MockFunction]}
      red={false}
      style={Object {}}
    >
      Close
    </ButtonGhost>
  </Stack>
</ModalToast>
`;
